### DISCUSSION TITLE
Help me understand a little better please

### DISCUSSION THREAD

**Discussion by Node815** [slackhq/nebula#1078] [Q&A] [answered]
For Nebula, do you have to open a port for the Lighthouse or Relay in your firewall?   
As I understand, you need a Lighthouse to act as your director (reverse proxy?) or is it relay?   Or maybe both? 

What my endgame is, would be to connect my vanity domain to use it and have precisely 3 services publicly exposed (IDP, Home Assistant and a Vaultwarden)

**Comment by johnmaguire** [comment, marked as answer]
Hi @Node815 - a [Lighthouse](https://docs.defined.net/glossary/lighthouse/) acts sort of like a directory for your other nodes, informing `nodeA` how it can find `nodeB`. A [relay](https://docs.defined.net/glossary/relay/) can relay traffic between two nodes that can't communicate directly, for whatever reason. You may use your Lighthouse as a relay by setting `am_relay: true` and listing it in the [`relay.relays`](https://nebula.defined.net/docs/config/relay/) section of your nodes.

Lighthouses require a static, port opened in your firewall, so that nodes are able to connect to it directly. It is recommended that you also use a static IP, however you can define them in your config using DNS names.

It is also advisable that your relay also has a static, open port in your firewall to ensure all of your nodes can access it.

**Comment by Node815** [comment]
Thanks, I will pass then - I don't want to have any open ports.   Just a personal policy of mine.
